sdk/dotnet/Pulumi/Resources/DependencyProviderResource.cs

```diff
@@ -32,5 +26,29 @@ public DependencyProviderResource(string reference)
             var idData = OutputData.Create(resources, id, isKnown: true, isSecret: false);
             this.Id = new Output<string>(Task.FromResult(idData));
         }
+
+        private static string GetPackage(string reference)
+        {
+            var (urn, _) = ParseReference(reference);
+            var urnParts = urn.Split("::");
```

Comment:
We really should consolidate the URN parsing in Node.js, Python, and .NET, similar to [what we have in Go](https://github.com/pulumi/pulumi/blob/master/sdk/go/common/resource/urn.go). I considered doing that as part of this, but it was looking like a larger change than I wanted to bite off.